pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -210,15 +213,31 @@ public String listTableConfigs(@ApiParam(value = "realtime|offline") @QueryParam
 
       if (tableType == null) {
         tableNames = _pinotHelixResourceManager.getAllRawTables();
+      } else if (tableType == TableType.REALTIME) {
+        tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
       } else {
-        if (tableType == TableType.REALTIME) {
-          tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
-        } else {
-          tableNames = _pinotHelixResourceManager.getAllOfflineTables();
-        }
+        tableNames = _pinotHelixResourceManager.getAllOfflineTables();
       }
 
-      Collections.sort(tableNames);
+      if (sortType == "name") {
+        // Sort table names alphabetically
+        Collections.sort(tableNames, sortAsc ? null : Collections.reverseOrder());
+      } else {
+        // Sort table names based on (1) Create Time
+        TableType finalTableType = tableType;
+        Collections.sort(tableNames, (Comparator<String>) (o1, o2) -> {
+          TableStats s1, s2;
+          s1 = _pinotHelixResourceManager.getTableStats(o1);
```

Comment:
Don't read table stats in the comparator. Cache them before the sorting.

Reply:
I am using a hashmap, does it make sense?